pkg/kwok/controllers/node_controller.go

```diff
@@ -389,10 +394,14 @@ func (c *NodeController) preprocess(ctx context.Context, node *corev1.Node) erro
 		)
 	}
 
+	// We initialize a normal(fresh) job with a zero `RetryCount`.
+	// The counter may increase if the job fails to apply.
+	retryTimes := 0
 	item := resourceStageJob[*corev1.Node]{
-		Resource: node,
-		Stage:    stage,
-		Key:      key,
+		Resource:   node,
+		Stage:      stage,
+		Key:        key,
+		RetryCount: &retryTimes,
 	}
 
 	c.delayQueue.AddAfter(item, delay)
```

Comment:
Maybe we should wrap the following into methods and try to de-duplicate multiple events on the same object

``` golang
c.delayQueue.AddWeightAfter(...)
c.delayQueueMapping.Store(...)
```